### PR TITLE
Features/debtors screen

### PR DESCRIPTION
**Pull request by santicarmo31**
# About this PR 
> Remove all templates marks before sending a PR or it will be rejected

I have implemented the Debtors screen design. 

### Evidence

Please find below the attached image.

simulator screen shot - iphone 8 plus - 2018-02-26 at 13 42 40

### Release Notes 

 - Issue: Debtors screen #26
 
### Useful information

I have added the function `go` to the `ViewControllerProtocol` extension in order to don't need to add it to every `ViewController` which conforms the `View` protocol

### CODE REVIEW

File: blacklist/blacklist/Common/Utils/SegueIndentifier.swift [hunk 1]
```diff
@@ -10,5 +10,6 @@ import Foundation
 
 enum SegueIndentifier: String {
     // Add your segues identifier here as enum, use .rawValue to use it
-    case SegueTest
+    case segueTest = "Test"
```
Comment (molayab) [inline blacklist/blacklist/Common/Utils/SegueIndentifier.swift]: You must not set a value to a string enum, the case is a string itself, please remove those values and try to use only the enum identifiers with the corresponding rawValue.

File: blacklist/blacklist/Common/Utils/StyleSheet.swift [hunk 2]
```diff
@@ -10,10 +10,10 @@ import UIKit
...
 struct StyleSheet {
     // Define here fonts, colors and design constants.
-    struct Font {
-        static let robotoRegular = "Roboto-Regular"
-        static let robotoLight = "Roboto-Light"
-        static let lobsterRegular = "Lobster-regular"
+    enum Font: String {
+        case robotoRegular = "Roboto-Regular"
+        case robotoLight = "Roboto-Light"
+        case lobsterRegular = "Lobster-regular"
```
Comment (molayab) [inline blacklist/blacklist/Common/Utils/StyleSheet.swift]: In this case, we must declare an explicit rawValue because the identifier cannot have special characters. Do not modify this enum.

File: blacklist/blacklist/Common/Utils/StyleSheet.swift [hunk 3]
```diff
@@ -29,5 +29,10 @@ struct StyleSheet {
             static let emptyHeaderText = UIColor(hex: 0x222831)
             static let emptySubHeaderText = UIColor(hex: 0x222831)
         }
+
+        struct Debtors {
+            static let background = #colorLiteral(red: 0.864806354, green: 0.8960095048, blue: 0.8077570796, alpha: 1)
```
Comment (molayab) [inline blacklist/blacklist/Common/Utils/StyleSheet.swift]: Try to avoid literal colors, is great in Xcode but we have to try to avoid macros. Use the hexadecimal representation instead. `UIColor(hex: 0xFF00FF)`

File: blacklist/blacklist/Scenes/BlackListTabBar/BlackListTabBarPresenter.swift [hunk 4]
```diff
@@ -0,0 +1,20 @@
+//
```
Comment (molayab) [inline blacklist/blacklist/Scenes/BlackListTabBar/BlackListTabBarPresenter.swift]: Why is this file necessary?

File: blacklist/blacklist/Scenes/Debtors/CustomViews/DebtorCollectionViewCell.swift [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    @IBOutlet weak var monsterImageView: UIImageView!
+    @IBOutlet weak var debtorNameLabel: UILabel!
+    @IBOutlet weak var monsterContainerView: UIView!
+    @IBOutlet weak var checkMarkImageView: UIImageView!
+
+    // MARK: - Vars & Constants
+
+    static var identifier: String {
+        return String(describing: self)
```
Comment (molayab) [inline blacklist/blacklist/Scenes/Debtors/CustomViews/DebtorCollectionViewCell.swift]: Great solution 🥇

File: blacklist/blacklist/Scenes/Debtors/DebtorsPresenter.swift [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+class DebtorsPresenter: Presenter {
+    typealias V = DebtorsView
+
+    weak var view: V?
+
+    func loadDebtors() {
+        // TODO: Call Realm or whatever to get the debtors list
+
+        var debtors: [Debtor] = [
```
Comment (molayab) [inline blacklist/blacklist/Scenes/Debtors/DebtorsPresenter.swift]: We can try to add some mock data, whatever, the main idea is to populate realm on the next tasks

File: blacklist/blacklist/Scenes/Debtors/DebtorsRouter.swift [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+//  Copyright © 2018 Opelty's Open Source Projects. All rights reserved.
+//
+
+import UIKit
+
+class DebtorsRouter: Router {
+    // Add transition methods here...
+
+    func goToContinueAction() { // TODO: Finish this
+        print("Going to continue screen...")
```
Comment (molayab) [inline blacklist/blacklist/Scenes/Debtors/DebtorsRouter.swift]: Please remove this `print` we can use it to indicate something but when the main feature has implemented please remove it in order to avoid wasted info in the console.

File: blacklist/blacklist/Common/Utils/SegueIndentifier.swift (hunk 1 above)
Comment (santicarmo31) [reply blacklist/blacklist/Common/Utils/SegueIndentifier.swift]: How I have forgotten that. Thanks, men.

File: blacklist/blacklist/Scenes/BlackListTabBar/BlackListTabBarPresenter.swift (hunk 4 above)
Comment (santicarmo31) [reply blacklist/blacklist/Scenes/BlackListTabBar/BlackListTabBarPresenter.swift]: Because we are using the router and the `configure` function needs a presenter because it is not an optional parameter @molayab
